src/udp.rs

```diff
@@ -316,7 +590,8 @@ where
     HostImpl: Host,
 {
     fn bind(&mut self, socket: &mut Self::UdpSocket, local_port: u16) -> Result<(), Self::Error> {
-        socket.open(&mut self.bus, local_port)?;
+        socket.set_port(&mut self.bus, local_port)?;
+        socket.open(&mut self.bus)?;
         Ok(())
     }
```

Comment:
@ryan-summers bind() is where we set the port to a give local_port and open the socket itself.